stdlib/Makefile

```diff
@@ -266,7 +266,7 @@ GNUISH_SED = \
 
 .INTERMEDIATE: .depend.tmp
 
-.depend.tmp:
+.depend.tmp: $(wildcard *.mli *.ml)
```

Comment:
This doesn't work (as CI is demonstrating) as it introduces an actual dependency between `.depend` on the files. It would alas appear I didn't test `make alldepend` in enough detail apropos https://github.com/ocaml/ocaml/pull/11126#pullrequestreview-917494259.

The fix is to change L275 below to:

```make
depend: .depend.tmp .depend
```

That means that when `Makefile` is loaded "normally" (e.g. `make -C stdlib all`), then the dependency of `.depend` on `.depend.tmp` is ignored (as `.depend.tmp` is intermediate) but when `make -C stdlib depend` is called, `.depend.tmp` is always built (because it's a direct dependency) but is still deleted the build (because `.INTERMEDIATE` files always are).

It's become a bit convoluted, though, just to avoid the explicit call to `rm -f .depend.tmp`, @shindere?